pkg/util/round_trippers.go

```diff
@@ -0,0 +1,50 @@
+package util
+
+import (
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"k8s.io/client-go/transport"
+)
+
+type proxyHeaderRoundTripper struct {
+	proxyHeaders http.Header
+	roundTripper http.RoundTripper
+}
+
+var _ http.RoundTripper = &proxyHeaderRoundTripper{}
+
+// NewProxyHeaderRoundTripperWrapperConstructor returns a RoundTripper wrapper that's usable within restConfig.WrapTransport.
+func NewProxyHeaderRoundTripperWrapperConstructor(wt transport.WrapperFunc, headers map[string]string) transport.WrapperFunc {
+	return func(rt http.RoundTripper) http.RoundTripper {
+		if wt != nil {
+			rt = wt(rt)
+		}
+		return &proxyHeaderRoundTripper{
+			proxyHeaders: parseProxyHeaders(headers),
+			roundTripper: rt,
+		}
+	}
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (r *proxyHeaderRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if tr, ok := r.roundTripper.(*http.Transport); ok {
+		tr.ProxyConnectHeader = r.proxyHeaders
```

Comment:
I think a `clone Transport` should be used here because the default value of Transport is `http.DefaultTransport`.

Reply:
like :
``` go
tr = tr.Clone()
```

Reply:
> I think a clone Transport should be used here because the default value of Transport is http.DefaultTransport.



I'm still reviewing.

Reply:
I still don't understand why we need clone, could you explain more clearly?

Reply:
https://github.com/golang/go/blob/master/src/net/http/client.go#L129-L142

Reply:
>	// RoundTrip should not modify the request, except for
	// consuming and closing the Request's Body. RoundTrip may
	// read fields of the request in a separate goroutine. Callers
	// should not mutate or reuse the request until the Response's
	// Body has been closed.

Reply:
I'm afraid not. `tr` is a temporary variable and will not be used outside of the if judge.

Reply:
> I'm afraid not. tr is a temporary variable and will not be used outside of the if judge.



It may be that I didn't describe it clearly. What I'm concerned about is the following situation.


![image](https://github.com/karmada-io/karmada/assets/17962021/81cfa5ea-4ac9-4e88-887f-b6c513c334e6)

Reply:
I think I understand what @chaunceyjiang  means. You are afraid that the value of r.roundTripper will be modified.

Reply:
I got it. I agree with you.